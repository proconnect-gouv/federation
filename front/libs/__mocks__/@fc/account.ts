import React from 'react';

export const AccountProvider = jest.fn();

export const AccountContext = React.createContext(expect.any(Object));
