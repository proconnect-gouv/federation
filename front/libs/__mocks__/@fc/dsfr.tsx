export const FranceConnectButton = jest.fn(() => <div>FranceConnectButton</div>);

export const SimpleButton = jest.fn(() => <div>SimpleButton</div>);

export const CheckboxInput = jest.fn(() => <div>CheckboxInput</div>);

export const ToggleInput = jest.fn(() => <div>ToggleInput</div>);

export const ApplicationLayout = jest.fn(() => <div>ApplicationLayout</div>);

export const LinkComponent = jest.fn(() => <div>LinkComponent</div>);

export const SearchBarComponent = jest.fn(() => <div>SearchBarComponent</div>);

export const Sizes = {
  LARGE: 'lg',
  MEDIUM: 'md',
  SMALL: 'sm',
};

export const AlertTypes = {
  ERROR: 'error',
  INFO: 'info',
  SUCCESS: 'success',
  WARNING: 'warning',
};

export const AlertComponent = jest.fn(() => <div>Alert</div>);

export const PaginationComponent = jest.fn(() => <div>PaginationComponent</div>);

export { ButtonTypes, IconPlacement, Priorities } from '@fc/dsfr';
