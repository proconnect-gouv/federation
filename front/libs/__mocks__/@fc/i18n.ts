export const I18nService = {
  initialize: jest.fn(),
};

export const t = jest.fn((key) => key);
