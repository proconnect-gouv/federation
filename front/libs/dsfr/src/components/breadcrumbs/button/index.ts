export * from './breadcrumbs-toggle.button';
