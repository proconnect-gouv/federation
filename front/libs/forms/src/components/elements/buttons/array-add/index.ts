export * from './array-add.button';
