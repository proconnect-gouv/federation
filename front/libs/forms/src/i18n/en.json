{
  "Form.button.copy": "Copier",
  "Form.submit": "Submit",
  "Form.reset": "Reset the form",
  "Form.message.valid": "Validation text",
  "Form.message.required": "This field is required",
  "Form.message.requestFailed": "The message could not be sent",
  "Form.message.requiredFields": "All fields with an asterisk are required.",
  "Form.select.placeholder": "Select an option",
  "Form.textarea.maxlength": " / {maxLength} characters",
  "Form.multiple.add": "Add a new value",
  "Form.multiple.remove": "Remove this value"
}
