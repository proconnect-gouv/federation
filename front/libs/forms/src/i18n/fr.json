{
  "Form.button.copy": "Copier",
  "Form.submit": "Envoyer ces informations",
  "Form.reset": "Effacer le formulaire",
  "Form.message.valid": "Texte de validation",
  "Form.message.required": "Ce champs est obligatoire",
  "Form.message.requestFailed": "Le message n’a pas pu être envoyé",
  "Form.message.requiredFields": "Tous les champs avec un astérisque sont obligatoires.",
  "Form.select.placeholder": "Sélectionner une option",
  "Form.textarea.maxlength": " / {maxLength} caractères",
  "Form.FORM_ERROR": "Une erreur s'est produite lors de la soumission du formulaire.",
  "Form.multiple.add": "Ajouter une valeur",
  "Form.multiple.remove": "Supprimer cette valeur"
}
