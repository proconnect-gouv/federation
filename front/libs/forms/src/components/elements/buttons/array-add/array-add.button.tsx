import React from 'react';

import type { PropsWithOnClick } from '@fc/common';
import { IconPlacement, Priorities, SimpleButton } from '@fc/dsfr';
import { t } from '@fc/i18n';

interface ArrayAddButtonProps extends Required<PropsWithOnClick> {
  disabled?: boolean;
  dataTestId?: string;
}

export const ArrayAddButton = React.memo(
  ({ dataTestId = undefined, disabled = false, onClick }: ArrayAddButtonProps) => (
    <SimpleButton
      className="fr-mt-2v"
      dataTestId={dataTestId}
      disabled={disabled}
      icon="add-line"
      iconPlacement={IconPlacement.LEFT}
      priority={Priorities.TERTIARY}
      title={t('Form.multiple.add')}
      onClick={onClick}>
      {t('Form.multiple.add')}
    </SimpleButton>
  ),
);

ArrayAddButton.displayName = 'ArrayAddButton';
