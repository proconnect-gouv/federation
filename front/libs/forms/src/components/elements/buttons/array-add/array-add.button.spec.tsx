import { render } from '@testing-library/react';

import { SimpleButton } from '@fc/dsfr';
import { t } from '@fc/i18n';

import { ArrayAddButton } from './array-add.button';

describe('ArrayAddButton', () => {
  it('should match the snapshot', () => {
    // Given
    const onClickMock = jest.fn();
    jest
      .mocked(t)
      .mockReturnValueOnce('Form.multiple.add')
      .mockReturnValueOnce('Form.multiple.add');

    // When
    const { container } = render(
      <ArrayAddButton dataTestId="any-data-testid-mock" onClick={onClickMock} />,
    );

    // Then
    expect(container).toMatchSnapshot();
    expect(SimpleButton).toHaveBeenCalledOnce();
    expect(SimpleButton).toHaveBeenCalledWith(
      {
        children: 'Form.multiple.add',
        className: 'fr-mt-2v',
        dataTestId: 'any-data-testid-mock',
        disabled: false,
        icon: 'add-line',
        iconPlacement: 'left',
        onClick: onClickMock,
        priority: 'tertiary',
        title: 'Form.multiple.add',
      },
      {},
    );
    expect(t).toHaveBeenCalledTimes(2);
    expect(t).toHaveBeenNthCalledWith(1, 'Form.multiple.add');
    expect(t).toHaveBeenNthCalledWith(2, 'Form.multiple.add');
  });

  it('should match the snapshot, when the button is disabled', () => {
    // Given
    const onClickMock = jest.fn();

    // When
    const { container } = render(<ArrayAddButton disabled onClick={onClickMock} />);

    // Then
    expect(container).toMatchSnapshot();
    expect(SimpleButton).toHaveBeenCalledOnce();
    expect(SimpleButton).toHaveBeenCalledWith(
      {
        children: 'Form.multiple.add',
        className: 'fr-mt-2v',
        disabled: true,
        icon: 'add-line',
        iconPlacement: 'left',
        onClick: onClickMock,
        priority: 'tertiary',
        title: 'Form.multiple.add',
      },
      {},
    );
    expect(t).toHaveBeenCalledTimes(2);
    expect(t).toHaveBeenNthCalledWith(1, 'Form.multiple.add');
    expect(t).toHaveBeenNthCalledWith(2, 'Form.multiple.add');
  });
});
