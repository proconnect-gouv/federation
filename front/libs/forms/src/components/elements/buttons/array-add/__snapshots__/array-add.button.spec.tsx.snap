// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrayAddButton should match the snapshot 1`] = `
<div>
  <div>
    SimpleButton
  </div>
</div>
`;

exports[`ArrayAddButton should match the snapshot, when the button is disabled 1`] = `
<div>
  <div>
    SimpleButton
  </div>
</div>
`;
